lib/GLPI/Agent/Task/Inventory/Virtualization/Qemu.pm

```diff
@@ -47,6 +47,11 @@ sub _parseProcessList {
             $values->{uuid} = $1;
         } elsif ($option =~ m/^enable-kvm/) {
             $values->{vmtype} = "kvm";
+        } elsif ($option =~ m/^smp ([^\s,]+)/) {
+            my ($cpus) = split(/,/, $1);
+            if ($cpus =~ /^\d+$/) {
+                $values->{vcpu} = int($cpus);
+            }
```

Comment:
Reading again this condition, I'm perplex. To me the regexp on `$option` means you take everything not a space or a comma after `smp` option. And after you're splitting against comma (which shouldn't be there) so. I found also in qemu help that the number can be set with: `cpus=2` for 2 cpus as example. Then I think a better code here would be:
```suggestion
        } elsif ($option =~ m/^smp (\S+)$/) {
            my ($cpus) = grep { /^(?:cpus=)?\d+$/ } split(/,/, $1);
            if ($cpus && $cpus =~ /(\d+)$/) {
                $values->{vcpu} = int($1);
            }
```
This code will also support to set `cpus=2` not at first place.

Reply:
Sounds reasonable. But sould we think a bit wider? what about:
`-smp cores=2,threads=2,sockets=1` How did GLPI define CPU?